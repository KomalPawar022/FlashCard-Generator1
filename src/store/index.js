import { configureStore } from "@reduxjs/toolkit";
import groupReducer from "./slices/group-slice";
import cardReducer from "./slices/card-slice";

const store = configureStore({
  reducer: {
    cardGroup: groupReducer,
    card: cardReducer,
  },
});
export default store;
