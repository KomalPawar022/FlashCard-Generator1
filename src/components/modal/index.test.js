import { render } from "@testing-library/react";
import { describe, expect, it } from "@jest/globals";
import Modal from "./index";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../../store";

function renderWithRedux(ui, store1 = store) {
  return render(
    <BrowserRouter>
      <Provider store={store1}>{ui}</Provider>
    </BrowserRouter>,
  );
}

describe("renders modal correctly", () => {
  it("should match the snapshot", () => {
    const { asFragment } = renderWithRedux(
      <Modal body={"Test"} onClose={() => {}} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
