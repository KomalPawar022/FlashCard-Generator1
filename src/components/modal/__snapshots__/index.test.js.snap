// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal correctly should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class=" night-mode-container z-1 fixed pt-[200px] top-0 left-0 w-full h-full justify-center items-center  bg-opacity-0 transition duration-150 ease-in-out"
  >
    <div
      class="relative bg-white m-auto p-0 shadow-2xl border-lime-600 border-4 w-[80vw] text-center text-black-900 duration-75  rounded-xl h-[200px] justify-center items-center"
    >
      <div
        class="bg-white-900 text-center text-black-900 justify-center items-center"
      />
      <nav>
        <span
          class="night-mode-container float-right text-2xl cursor-pointer mr-5"
        >
          <a
            href="/my-flashcards"
          >
            âœ–
          </a>
        </span>
      </nav>
      <div
        class="flex flex-col justify-center items-center m-2 bg-white-900 mt-15 h-full"
      >
        <h1
          class="font-bold text-2xl"
        >
           Test
        </h1>
      </div>
      <div
        class="h-[10px]"
      />
    </div>
  </div>
</DocumentFragment>
`;
